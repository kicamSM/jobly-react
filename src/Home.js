import React from "react";
import { Card, CardBody, CardTitle } from "reactstrap";
import './Home.css'

/**
 * Display home page and count of snacks and drinks
 *
 */

function Home() {
  return (
      <section className="col-md-8 Home">
        <Card>
          <CardTitle>
              <div className="font-weight-bold Home-div">
                <h1>Jobly</h1>
              </div>
          </CardTitle>
          <CardBody className="text-center">
            <div>All the jobs in one, convenient plae.</div>
            <h2>Welcome Back</h2>
          </CardBody>
        </Card>
      </section>
  );
}

export default Home;
