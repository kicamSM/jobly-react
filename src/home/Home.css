body {
    background-color: lightblue;
}

/* a {
    background-color: blue;
    color: white;
    text-decoration: none;
  } */


.Home-Link {
    background-color: #24a0ed;
    color: white;
    font-size: 20px;
    padding: 5px;
    border: gray solid 1px;
    text-decoration: none;
    border-radius: 5px;
}