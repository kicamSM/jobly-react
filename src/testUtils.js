import React from "react";
import UserContext from "./repeated/UserContext";

const demoUser = {
  username: "testuser",
  firstName: "testfirst",
  lastName: "testlast",
  email: "test@test.net",
  photo_url: null,
};

const UserProvider =
    ({ children, user = demoUser, apply = () => false }) => (
    <UserContext.Provider value={{ user, apply }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };
