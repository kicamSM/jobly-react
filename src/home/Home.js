import React, { useContext } from "react";
import UserContext from "../repeated/UserContext";
import './Home.css'
import { Link } from "react-router-dom";
import { 
  Card, 
  CardBody, 
  CardText, 
  CardTitle,
  Button
} from "reactstrap";

/**
 * Display home page
 *
 */

function Home() {

  /** Get user from context*/

  const { user } = useContext(UserContext);

  /** Render page */

    return (
      <section className="col-md-8 Home">
        <Card>
          <CardTitle>
                <h1>Jobly</h1>
          </CardTitle>
          <CardBody >
            <CardText>All the jobs in one, convenient place.</CardText>
            {user && <h2>Welcome Back, {user.firstName}!</h2>}
            { !user && ( <> <CardText>
              <Link className="Home-Link" to="/login"><Button>Login</Button></Link>
              <Link className="Home-Link" to="/signup"><Button>Signup</Button></Link>
            </CardText>
            </>)}
          </CardBody>
        </Card>
      </section>
    )
}

export default Home;
