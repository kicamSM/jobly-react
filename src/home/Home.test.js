import React from "react";
import { render, screen } from "@testing-library/react";
import Home from "./Home";
import { MemoryRouter } from "react-router-dom";
import { UserProvider } from "../testUtils";

test("it should render when logged out", () => {
  const { asFragment } = render(
    <MemoryRouter initialEntries={["/signup"]}>
      <UserProvider user={null}>
          <Home />
      </UserProvider>
    </MemoryRouter>,
  );
  expect(screen.getByText('Jobly')).toBeInTheDocument();
  expect(screen.getByText('All the jobs in one, convenient place.')).toBeInTheDocument();
  expect(screen.getByText('Login')).toBeInTheDocument();
  expect(screen.getByText('Signup')).toBeInTheDocument();
}); 

it("matches snapshot when logged out", function() {
  const { asFragment } = render(
    <MemoryRouter initialEntries={["/signup"]}>
      <UserProvider user={null}>
          <Home />
      </UserProvider>
    </MemoryRouter>,
  );
    expect(asFragment()).toMatchSnapshot();
});
  

test("it should render when logged in", () => {
  const { asFragment } = render(
    <MemoryRouter initialEntries={["/signup"]}>
      <UserProvider>
          <Home />
      </UserProvider>
    </MemoryRouter>,
  );
  expect(screen.getByText('Jobly')).toBeInTheDocument();
  expect(screen.getByText('All the jobs in one, convenient place.')).toBeInTheDocument();
  expect(screen.getByText('Welcome Back, testfirst!')).toBeInTheDocument();
}); 

it("matches snapshot when logged in", function() {
  const { asFragment } = render(
    <MemoryRouter initialEntries={["/signup"]}>
      <UserProvider>
          <Home />
      </UserProvider>
    </MemoryRouter>,
  );
    expect(asFragment()).toMatchSnapshot();
});
  