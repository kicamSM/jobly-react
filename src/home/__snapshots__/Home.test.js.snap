// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="col-md-8 Home"
  >
    <div
      class="card"
    >
      <div
        class="card-title"
      >
        <h1>
          Jobly
        </h1>
      </div>
      <div
        class="card-body"
      >
        <p
          class="card-text"
        >
          All the jobs in one, convenient place.
        </p>
        <h2>
          Welcome Back, testfirst!
        </h2>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`matches snapshot when logged in 1`] = `
<DocumentFragment>
  <section
    class="col-md-8 Home"
  >
    <div
      class="card"
    >
      <div
        class="card-title"
      >
        <h1>
          Jobly
        </h1>
      </div>
      <div
        class="card-body"
      >
        <p
          class="card-text"
        >
          All the jobs in one, convenient place.
        </p>
        <h2>
          Welcome Back, testfirst!
        </h2>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <section
    class="col-md-8 Home"
  >
    <div
      class="card"
    >
      <div
        class="card-title"
      >
        <h1>
          Jobly
        </h1>
      </div>
      <div
        class="card-body"
      >
        <p
          class="card-text"
        >
          All the jobs in one, convenient place.
        </p>
         
        <p
          class="card-text"
        >
          <a
            class="Home-Link"
            href="/login"
          >
            <button
              class="btn btn-secondary"
            >
              Login
            </button>
          </a>
          <a
            class="Home-Link"
            href="/signup"
          >
            <button
              class="btn btn-secondary"
            >
              Signup
            </button>
          </a>
        </p>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
